### PR TITLE
tbvcfreport changes, part 1

### PR DESCRIPTION
**Pull request by pvanheus**
* Speed up DB queries
* Add SNP-IT support for lineage info

### CODE REVIEW

File: tbvcfreport/templates/base.html [hunk 1]
```diff
@@ -19,7 +19,33 @@
             <h1 class="mt-5">
                 {{ file_name }} Report
             </h1>
-            <hr> {% block content %} {% endblock %}
+            <hr>
+            {% if 'lineage' in data and data['lineage']['lineage'] %}
```
Comment (thobalose) [inline tbvcfreport/templates/base.html:23]: @pvanheus Kindly make this a separate report.

File: tbvcfreport/tbvcfreport.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
 """
 Interface to CLI
 """
+import json
 import logging
 import os
 import sys
 
 import click
+from snpit import snpit
```
Comment (thobalose) [inline tbvcfreport/tbvcfreport.py:10]: @pvanheus Travis is failing due to snpit not being in the `requirements.txt`. You can install this via `pip install -e git+https://github.com/samlipworth/snpit.git@V1.1#egg=snpit` and doing a `pip freeze`

File: setup.py [hunk 3]
```diff
@@ -18,11 +18,11 @@
         'tbvcfreport': ['templates/*.html'],
     },
     install_requires=[
-        'click',
-        'py2neo',
-        'Jinja2',
-        'tqdm',
-        'PyVCF',
+        # 'click',
```
Comment (thobalose) [inline setup.py]: @pvanheus Do not forget to uncomment the requirements.

File: tbvcfreport/tbvcfreport.py [hunk 4]
```diff
@@ -55,14 +57,90 @@ def generate(vcf_dir):
...
-                    [os.path.abspath(vcf_dir), vcf_file]
-                )
-                if check_vcf(vcf_file):
-                    file_name = vcf_file.split('/')[-1].split('.')[0]
-                    vcf_file = VCFProc(vcf_file=vcf_file)
-                    variants = vcf_file.parse()
-                    generate_report(file_name=file_name, data=variants)
+                (base, ext) = os.path.splitext(vcf_file)
+                print(vcf_file)
+                if ext == '.vcf':
```
Comment (thobalose) [inline tbvcfreport/tbvcfreport.py]: @pvanheus A reminder to move this to https://github.com/COMBAT-TB/tbvcfreport/blob/669311bcb23673d2456e37ebeea724b0a33aec64/tbvcfreport/tbvcfreport.py#L144

**Comment by thobalose** [comment]
@pvanheus 

We also need to update https://github.com/COMBAT-TB/tbvcfreport/blob/master/galaxy/tbvcfreport.xml to cater for multiple inputs (json and vcf).

**Review by thobalose — CHANGES_REQUESTED** [review]

File: tbvcfreport/templates/base.html (hunk 1 above)
Comment (pvanheus) [reply tbvcfreport/templates/base.html:23]: done.

File: tbvcfreport/tbvcfreport.py (hunk 2 above)
Comment (pvanheus) [reply tbvcfreport/tbvcfreport.py:10]: import added to setup.py via `dependency_links` as suggested.

File: tbvcfreport/tbvcfreport.py (hunk 4 above)
Comment (pvanheus) [reply tbvcfreport/tbvcfreport.py]: done.

**Comment by pvanheus** [comment]
Strange error at the end:

```
usage: anaconda [-h] [--disable-ssl-warnings] [--show-traceback] [-v] [-q]
                [-V] [-t TOKEN] [-s SITE]
                ...
anaconda: error: argument : invalid choice: 'thoba' (choose from 'auth', 'label', 'channel', 'config', 'copy', 'download', 'groups', 'login', 'logout', 'move', 'notebook', 'package', 'remove', 'search', 'show', 'upload', 'whoami')
```

**Pull request merged**